modules/iopcore/cdvdman/device-hdd.c

```diff
@@ -77,6 +79,19 @@ void DeviceInit(void)
     NumParts = apaHeader.num_partitions;
 }
 
+void DeviceSetupZSO(u8 *buffer)
+{
+    ziso_read_sector(buffer, 16, 1);
+    u32 maxLBA = *(u32 *)(buffer + 80);
+    if (maxLBA > 0 && maxLBA < ziso_total_block) { // dual layer check
+        if (ziso_read_sector(buffer, maxLBA, 1) == 1) {
+            if ((buffer[0x00] == 1) && (!strncmp(&buffer[0x01], "CD001", 5))) {
+                cdvdman_settings.common.layer1_start = maxLBA - 16; // adjust second layer start
+            }
+        }
+    }
+}
```

Comment:
The block device code should not have ZSO specific code in it. This code only depends on 'ziso_read_sector' and can be moved to the higher level zso code (generic for all devices).

Other devices do not need `cdvdman_settings.common.layer1_start` to be adjusted?

Reply:
Well, only HDD needed this adjustment, and I thought maybe allowing devices to have their own custom handlers for when ZSO is initialized might be useful in future devices.
If the code ends up not being used we can move it to a higher layer, it shouldn't break other devices.